### PR TITLE
Issue #13345: Enable TestCases for GenericWhitespaceCheckExamplesTest

### PR DESCRIPTION
**Pull request by chrccl**
Issue #13345: Enable TestCases for GenericWhitespaceCheckExamplesTest

### CODE REVIEW

**Comment by chrccl** [comment]
Good Morning @romani ! Sorry, I had to close the previous PR for some strange error (It could not update it with another commit). I've done your reviews... But now I have some failed checks. Can you please help me? I'm struggling with this issue.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+// xdoc section -- start
+public class Example1 {
+  @FunctionalInterface
+  interface ImmListBldr<T> {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:20]:
```suggestion
  interface ImmutableListBuilder<T> {
```

CI is failing from spelling check, we have no need to extend whitelist for this.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Collections.sort(new ArrayList<String>(), Comparable::<String>compareTo);
+    // Constructor call
+    MyClass<String, String> obj = new MyClass<String, String>();
+    List a = new ArrayList<>  (); // ok, until #14344
+  }
+}
+// xdoc section -- end
+class Box {
+  public static <T> Box of(T value) {
+    // Implementation details
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:56]:
```suggestion
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.google.common.collect.ImmutableList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+// xdoc section -- start
+public class Example1 {
+  @FunctionalInterface
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:19]:
```suggestion

  @FunctionalInterface
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  interface ImmListBldr<T> {
+      ImmutableList.Builder<T> createBuilder();
+  }
+  // Generic methods definitions
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:23]:
```suggestion

  // Generic methods definitions
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  public static <K, V> boolean foo(K key, V value) {
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:24]:
```suggestion
  public static <K, V> boolean isFoo(K key, V value) {
```

Please do for all similar methods, methods with boolean return values should nearly always be `is...` or `has...`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    return false;
+  }
+  public<T> void foo() {} // violation, ''<' is not preceded with whitespace.'
+  // Generic type definition
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:28]:
```suggestion

  // Generic type definition
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    public MyClass() {}
+    public MyClass(T1 t1, T2 t2) {}
+  }
+
+  public void examples() {
+    // Generic type reference
+    Map<String, List<Integer>> p;
+    Map<Integer, String>m; // violation, ''>' is followed by an illegal character.'
+    List< String> l; // violation, ''<' is followed by whitespace.'
+    // Generic preceded method name
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:39]:
```suggestion

    // Generic method call
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    boolean same = Example1.<Object, Object>foo(null, null);
+    Box b = Box. <String>of("foo"); // violation, ''<' is preceded with whitespace.'
+    // Diamond operator
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:42]:
```suggestion

    // Diamond operator
```

Also, I do not see any diamond operators in the following examples.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    MyClass<Integer, String> p1 = new MyClass<Integer, String>(1, "apple");
+    MyClass<Integer, Integer > p2; // violation, ''>' is preceded with whitespace.'
+    // Method reference
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:45]:
```suggestion

    // Method reference
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    ImmListBldr<String> builderFactory = ImmutableList.Builder<String>::new;
+    Collections.sort(new ArrayList<String>(), Comparable::<String>compareTo);
+    // Constructor call
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:48]:
```suggestion

    // Generic constructor call
```

File: src/xdocs/checks/whitespace/genericwhitespace.xml.template [hunk 11]
```diff
@@ -40,23 +40,15 @@
         </p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.txt"/>
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java"/>
           <param name="type" value="config"/>
         </macro>
         <p id="Example1-code">
-          Examples with correct spacing:
+          Examples with correct and uncorrect spacing:
```
Comment (nrmancuso) [inline src/xdocs/checks/whitespace/genericwhitespace.xml.template:47]:
```suggestion
          Example:
```
`uncorrect` is an uncommon/unused word, and we have no need for extra text here regardless.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    MyClass<String, String> obj = new MyClass<String, String>();
+    List a = new ArrayList<>  (); // ok, until #14344
+  }
+}
+// xdoc section -- end
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:53]:
```suggestion
// xdoc section -- end

```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  }
+  public<T> void foo() {} // violation, ''<' is not preceded with whitespace.'
+  // Generic type definition
+  class MyClass<T1, T2> {
+    public MyClass() {}
+    public MyClass(T1 t1, T2 t2) {}
+  }
+
+  public void examples() {
+    // Generic type reference
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:35]:
```suggestion
    // Generic type declaration
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 1 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 2 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 4 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:20]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 5 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 6 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:24]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 8 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:40]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 9 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:44]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 10 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java]: solved

File: src/xdocs/checks/whitespace/genericwhitespace.xml.template (hunk 11 above)
Comment (chrccl) [reply src/xdocs/checks/whitespace/genericwhitespace.xml.template]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 12 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:53]: solved

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 13 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java]: solved

**Comment by chrccl** [comment]
Hey @nrmancuso ,

Hope you're doing awesome! 🌟
Sorry for the silence – got caught up in the uni whirlwind. 🎓 But good news: I've tackled all the tweaks you wanted for the project!
Would love your thoughts on what's next. Any guidance from your end would be golden. 🙌
Thanks a ton for your patience!

**Comment by romani** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
ok to merge

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 8 above)
Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:40]: Not done, there is still no diamond operators here.

**Comment by chrccl** [comment]
> ok to merge

@romani ping

**Review by nrmancuso — CHANGES_REQUESTED** [review]
https://github.com/checkstyle/checkstyle/pull/14364#discussion_r1485723628

**Comment by chrccl** [comment]
> [#14364 (comment)](https://github.com/checkstyle/checkstyle/pull/14364#discussion_r1485723628)

Morning sir @nrmancuso, please check now.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 8 above)
Comment (chrccl) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:42]: done

**Comment by chrccl** [comment]
@romani @nrmancuso ping

**Comment by romani** [comment]
@chrccl , please switch to another issue or fixes we will handle this PR soon.

**Comment by romani** [comment]
GitHub, generate website

**Comment by chrccl** [comment]
> @chrccl , please switch to another issue or fixes we will handle this PR soon.

ok, sorry I read this message only after my last push... Sorry for the mess I created! I'm confident that with the upcoming issues, I will improve and contribute better to the project!!!

**Comment by romani** [comment]
@chrccl, please rebase on our latest code in master branch.
Please address failing CI https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=19707&view=logs&jobId=e663805c-ee13-5a24-f979-d725e0346c8e&j=e663805c-ee13-5a24-f979-d725e0346c8e&t=40ff6e43-c5d7-5753-f947-23012624b0c6

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Still need to complete this item, somehow we now lost all diamond operators. https://github.com/checkstyle/checkstyle/pull/14364/files#r1489472956

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java (hunk 8 above)
Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:41]: @chrccl please do this.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+    final C<Integer> a2 = new C <>();
+    // violation above, ''<' is preceded with whitespace.'
+
+    // Method reference
+    ListBuilder<String> builderFactory = ImmutableList.Builder<String>::new;
+    Collections.sort(new ArrayList<String>(), Comparable::<String>compareTo);
+
+    // Generic constructor call
+    MyClass<String, String> obj = new MyClass<String, String>();
+    List a = new ArrayList<>  (); // ok, until #14344
```
Comment (sktpy) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/Example1.java:52]: Fix is merged, you can update this input and the corresponding test to have a violation here.

**Comment by romani** [comment]
if no respnse in 2 weeks, PR will be marked as abandoned and might be closed.

**Comment by romani** [comment]
Merged at https://github.com/checkstyle/checkstyle/pull/14529

**Pull request closed without merging**